### PR TITLE
Fix to N notifications (N tabs open)

### PR DESCRIPTION
**Pull request by fernandog**
Code from @sharkykh

I did a quick test and it's not working
image

Fixes: https://github.com/pymedusa/Medusa/issues/2897

### CODE REVIEW

**Comment by fernandog** [comment]
@OmgImAlexis not working

image

File: static/js/ajax-notifications.js [hunk 1]
```diff
@@ -39,7 +42,7 @@ function wsCheckNotifications() {
 
         // Add handling for different kinds of events. For ex: {"event": "notification", "data": {"title": ..}}
         if (msg.event === 'notification') {
-            displayPNotify(msg.data.type, msg.data.title, msg.data.body);
+            displayPNotify(msg.data.type, msg.data.title, msg.data.body, msg.data.hash);
         } else {
             displayPNotify('info', '', msg);
```
Comment (OmgImAlexis) [inline static/js/ajax-notifications.js]: Does the notification you're testing have `msg.event === 'notification'`?

Comment (fernandog) [inline static/js/ajax-notifications.js]: how to check that?

Comment (fernandog) [inline static/js/ajax-notifications.js]: image

Comment (sharkykh) [inline static/js/ajax-notifications.js]: You can add `console.log(msg.event)` before line 44 so it shows up in the DevTools console

Comment (fernandog) [inline static/js/ajax-notifications.js]: here is the issue:

image

Comment (sharkykh) [inline static/js/ajax-notifications.js]: What does the response from /ui/get_messages give you? (Open the network tab, then generate a notification from Medusa)

Comment (fernandog) [inline static/js/ajax-notifications.js]: I added a break-point here but it's never reach this code:

image


maybe the websockets @p0psicles ?

Comment (sharkykh) [inline static/js/ajax-notifications.js]: I just figured it out.
You changed in the wrong location (this is not the websocket implementation)

You need to change it in [`medusa/ui.py`](https://github.com/pymedusa/Medusa/blob/feature/issue_2897/medusa/ui.py) in both [`Notifications.message()`](https://github.com/pymedusa/Medusa/blob/feature/issue_2897/medusa/ui.py#L47-L50) and [`Notifications.error()`](https://github.com/pymedusa/Medusa/blob/feature/issue_2897/medusa/ui.py#L59-L63)
(but you should leave this here, too)

Comment (OmgImAlexis) [inline static/js/ajax-notifications.js]: I was about to say that when your comment appeared. 😆

Comment (fernandog) [inline static/js/ajax-notifications.js]: worked! thanks

File: static/js/ajax-notifications.js [hunk 2]
```diff
@@ -15,10 +15,13 @@ PNotify.prototype.options.styling = 'jqueryui';
 PNotify.prototype.options.width = '340px';
 PNotify.desktop.permission();
 
-function displayPNotify(type, title, message) {
+function displayPNotify(type, title, message, id) {
```
Comment (p0psicles) [inline static/js/ajax-notifications.js]: In python its preferred to not use buildins like `id`. So you can easily solve this, by rename it to notification_id or something like that.

Comment (p0psicles) [inline static/js/ajax-notifications.js]: Ow and nice work!

Comment (sharkykh) [inline static/js/ajax-notifications.js]: @p0psicles This is Javascript?

Comment (p0psicles) [inline static/js/ajax-notifications.js]: ow lol. Your right. Don't know what i've been smoking.

**Pull request merged**